File: bundles/org.eclipse.vjet.vsf.vjo.kernel/src/org/eclipse/vjet/vsf/common/IDedupComparable.js
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
/**
* Tracks duplicate messages. <code>org.eclipse.vjet.vsf.service.DedupServiceHandler</code> 
* uses this as a comparator to check whether the two messages are duplicate.
*/
vjo.itype('org.eclipse.vjet.vsf.common.IDedupComparable')
.protos({
	/**
	* Checks whether the given message is allowed to be duplicate.
	*
	* @param {com.ebay.dsf.resource.html.event.handler.JsHandlerObjectEnum} message
	*        a message to be checked
	* @return {boolean}
	*        <code>true</code> if this message is not allowed to be duplicate
	*/
	//> public boolean shouldTrack(com.ebay.dsf.resource.html.event.handler.JsHandlerObjectEnum);
	shouldTrack : function (message) {},
	
	/**
	* Checks whether the two messages are duplicate.
	*
	* @param {com.ebay.dsf.resource.html.event.handler.JsHandlerObjectEnum} currentDsfMsg
	*        the current message to be compared
	* @param {com.ebay.dsf.resource.html.event.handler.JsHandlerObjectEnum} dsfMsg
	*        Another message to be compared from
	* @return {boolean}
	*        <code>true</code> if two messages are duplicate
	*/
	//> public boolean isDedup(com.ebay.dsf.resource.html.event.handler.JsHandlerObjectEnum,com.ebay.dsf.resource.html.event.handler.JsHandlerObjectEnum);
	isDedup : function (currentDsfMsg, dsfMsg) {}
})
.endType();

File: vsf/bundles/org.eclipse.vjet.vsf.vjo.kernel/src/org/eclipse/vjet/vsf/common/IDedupComparable.java
/*******************************************************************************
 * Copyright (c) 2012 eBay Inc. and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     eBay Inc. - initial API and implementation
 *******************************************************************************/
package org.eclipse.vjet.vsf.common;

import org.eclipse.vjet.dsf.javatojs.anno.AJsProxy;
import org.mozilla.mod.javascript.IJsJavaProxy;
import org.eclipse.vjet.vsf.resource.html.event.handler.JsHandlerObjectEnum;

//NativeJsProxy for vjo.dsf.common.IDedupComparable.js
@org.eclipse.vjet.dsf.resource.utils.CodeGen("NativeJsProxyGenerator")
@AJsProxy
public interface IDedupComparable extends IJsJavaProxy {

    boolean shouldTrack(JsHandlerObjectEnum message);

    boolean isDedup(JsHandlerObjectEnum currentDsfMsg, JsHandlerObjectEnum dsfMsg);
}